src/code/z_camera.c

```diff
@@ -22,9 +22,20 @@ typedef struct {
     Vec3s pos;
 } SplineData; // originally called SplinedatZ
 
-// 47 lines
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800437F0.s")
-// had an attempt at this but didn't get very close
+f32 func_800437F0(f32 arg0, f32 arg1) {
+    f32 percent = .4f;
+    f32 ret = fabsf(arg1);
+
+    if (arg0 < ret) {
+        ret = 1;
+    } else if (ret < (float)arg0 * (1 - percent)) { //! float cast needed to match
```

Comment:
we use f32 over float
```suggestion
    } else if (ret < (f32)arg0 * (1 - percent)) { //! float cast needed to match
```